### PR TITLE
fix: [CDS-34048]: update config file component to use single component when used in config form

### PR DESCRIPTION
**Pull request by willd0yle**
##### Summary:
Currently a field array is used for a single file when in the config form section. A new check has been added to allow a singe component to be rendered instead of a field array.
##### Screenshots:
Before:
Screenshot 2022-02-23 at 15 28 03

After:
Screenshot 2022-02-23 at 15 25 54

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by willd0yle** [comment]
retrigger titlecheck

**Comment by willd0yle** [comment]
trigger cypress

**Review by kylekstewart — APPROVED** [review]

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformArtifactoryForm.tsx [hunk 1]
```diff
@@ -233,67 +234,81 @@ export const TFArtifactoryForm: React.FC<StepProps<any> & TFArtifactoryProps> =
...
+                        label=""
+                        multiTextInputProps={{
+                          expressions,
+                          allowableTypes: allowableTypes.filter(item => item !== MultiTypeInputType.RUNTIME)
+                        }}
+                        style={{ width: 320 }}
+                      />
+                    ) : (
+                      <FieldArray
+                        name={tfArtifactoryFormInputNames(isConfig).artifactPaths}
```
Comment (chrisblakely01) [inline src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformArtifactoryForm.tsx:249]: Same Q as above

**Review by chrisblakely01 — APPROVED** [review]
1 Q but all good

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformArtifactoryForm.tsx [hunk 2]
```diff
@@ -233,67 +234,81 @@ export const TFArtifactoryForm: React.FC<StepProps<any> & TFArtifactoryProps> =
...
-                          <div>
-                            {map(selectedArtifacts, (path: PathInterface, index: number) => (
-                              <Layout.Horizontal
-                                key={`${path}-${index}`}
-                                flex={{ distribution: 'space-between' }}
-                                style={{ alignItems: 'end' }}
-                              >
+                    {isConfig ? (
+                      <FormInput.MultiTextInput
+                        name={`${tfArtifactoryFormInputNames(isConfig).artifactPaths}[0].path`}
```
[the remark this answers is not in the document]
Comment (willd0yle) [reply src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformArtifactoryForm.tsx:239]: There is not, everywhere this is used passes in `true` or `false` for the `isConfig`

**Review by YounixIO — APPROVED** [review]

**Pull request merged**